### ISSUE TITLE
Deprecate Rich text editor component

### ISSUE THREAD

**Issue by gregtyler** [ministryofjustice/moj-design-system#63]
[The Rich text editor component](https://moj-design-system.herokuapp.com/components/rich-text-editor) has significant accessibility and usability issues that we lack the resource to fix. Rich text editors are hard to get right, and we should use well-supported tools instead of trying to upkeep our own.

Whilst we'll continue to support the rich text editor in existing services, we will deprecate it to discourage any new implementations. Once sufficient time as passed, we will hopefully be able to remove it from the MOJ Design System.

The following steps should be taken, with suitable space between them to allow backpedaling or a change of direction:

- [ ] Add banner to documentation to discourage using the component (like the one on [Currency input](https://moj-design-system.herokuapp.com/components/currency-input)). We should point people towards fully supported rich text editors like TinyMCE
- [ ] Move the documentation page to a "legacy" section of the docs
- [ ] Remove the component from documentation completely
- [ ] Remove the component from the MOJ Design System
